src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -317,17 +316,13 @@ def get_src_py_file_to_dest_py_file_map(
         return src_py_file_to_dest_py_file_map
 
 
-def _create_new_sql_file_at_path(parent_dir: Path, file_name: str) -> Optional[Path]:
+def _generate_new_sql_file_name(parent_dir: Path, file_name: str) -> Optional[Path]:
     """
-    Creates an empty SQL file at the same location specified as the parent directory, with file_name as part of the new file's name.
+    Generates a SQL filename using the provided parent directory, with file_name as part of the new file's name.
     """
-    sql_file_name = f"__{file_name}_{uuid.uuid4().hex}.sql"
+    sql_file_name = f"__cli_codegen__{file_name}.sql"
```

Comment:
better for sure, but my favorite way to solve this is to reduce the problem to a single source of collisions: we introduce some kind of "generated" directory (we call it something like _generated) and add an option to the PDF to redefine it if need be, the same way you can change the deploy root. This way, most users don't need to worry about it, it's clean and easy to inspect. Diffs work. Filenames make sense. But if by any chance there's a collision, we tell the user something akin to "you've already used our directory, delete/rename it, or change it in your config", then we bail out of code generation altogether.

Reply:
Would `_generated` live at `output/deploy` ? With the current implementation, generated SQL files live next to their python files in the deploy root. If there is a `_generated` folder, then:
1. Do we essentially duplicate the path within `__generated` for the SQL file? e.g. if there is a `output/deploy/a/b/c.py`, will `c.sql` live at `output/deploy/__generated/a/b/c.sql`, or even ? 
2.  The user will need to inspect the `__generated` folder keeping the directory structure of `output/deploy` in mind. Might be too much to keep track of at once?
3. How will the upload work? Deploy root is a mirror of the stage, does that mean that the py file and the sql file will live at different paths in the stage?
4. I think depending on this, I'll implement the logic of the handler property, which is currently on hold.